pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,673 @@
+		scopes := []sdk.SecurityIntegrationScope{{Scope: "foo"}, {Scope: "bar"}}
+
+		secret, id := createSecretWithOAuthClientCredentialsFlow(t, integrationId, scopes, func(req *sdk.CreateWithOAuthClientCredentialsFlowSecretRequest) {
+			req.WithComment("a").
+				WithIfNotExists(true)
+		})
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSecretDetails(details, secretDetails{
+			Name:            id.Name(),
+			Comment:         "a",
+			SecretType:      "OAUTH2",
+			OauthScopes:     "[foo, bar]",
+			IntegrationName: integrationId.Name(),
+		})
+
+		assertSecret(t, secret, id, "OAUTH2", "a")
+	})
+
+	// It is possible to create secret without specifying both refresh token properties and scopes
+	// Scopes are not being inherited from the security_integration what is tested further
+	t.Run("Create: secretWithOAuth - minimal, without token and scopes", func(t *testing.T) {
+		secret, id := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{}, nil)
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSecretDetails(details, secretDetails{
+			Name:                        id.Name(),
+			SecretType:                  "OAUTH2",
+			OauthScopes:                 "",
+			OauthAccessTokenExpiryTime:  nil,
+			OauthRefreshTokenExpiryTime: nil,
+			IntegrationName:             integrationId.Name(),
+		})
+
+		assertSecret(t, secret, id, "OAUTH2", "")
+	})
+
+	// regarding the https://docs.snowflake.com/en/sql-reference/sql/create-secret secret should inherit security_integration scopes, but it does not do so
+	t.Run("Create: SecretWithOAuthClientCredentialsFlow - No Scopes Specified", func(t *testing.T) {
+		_, id := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{}, nil)
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		securityIntegrationProperties, _ := client.SecurityIntegrations.Describe(ctx, integrationId)
+		assert.Contains(t, securityIntegrationProperties, sdk.SecurityIntegrationProperty{Name: "OAUTH_ALLOWED_SCOPES", Type: "List", Value: "[foo, bar]", Default: "[]"})
+
+		assert.NotEqual(t, details.OauthScopes, securityIntegrationProperties)
+	})
+
+	t.Run("Create: SecretWithOAuthAuthorizationCodeFlow", func(t *testing.T) {
+		secret, id := createSecretWithOAuthAuthorizationCodeFlow(t, integrationId, "foo", refreshTokenExpiryTime, func(req *sdk.CreateWithOAuthAuthorizationCodeFlowSecretRequest) {
+			req.WithComment("a").
+				WithIfNotExists(true)
+		})
+
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSecretDetails(details, secretDetails{
+			Name:                        id.Name(),
+			SecretType:                  "OAUTH2",
+			Comment:                     "a",
+			OauthRefreshTokenExpiryTime: stringDateToSnowflakeTimeFormat(time.DateOnly, refreshTokenExpiryTime),
```

Comment:
@sfc-gh-jcieslak Do we have a helper function for comparing time? If not, this could be moved to helpers somewhere.

Reply:
Let's leave it as it is right now, as a helper function just for secrets.

Reply:
Left it as a helper for secret integration tests